[dynamic registrar] Free the return value from mono_signature_get_params and mono_signature_get_return_type.

rolfbjarne opened the pull request:
Before:

    There were 258042 MonoObjects created, 235166 MonoObjects freed, so 22876 were not freed. (dynamic registrar)
    There were 205804 MonoObjects created, 204219 MonoObjects freed, so 1585 were not freed. (static registrar)

After:

    There were 258066 MonoObjects created, 246781 MonoObjects freed, so 11285 were not freed. (dynamic registrar)
    There were 205804 MonoObjects created, 204491 MonoObjects freed, so 1313 were not freed. (static registrar)

vs-mobiletools-engineering-service2 commented:
### :fire: Tests failed catastrophically on Build (no summary found). :fire:

Result file $(TEST_SUMMARY_PATH) not found. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812737&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.12/4812737/sim/;/tests/vsdrops_index.html)
# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11730/4812737/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11730/4812737/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11730/4812737/apicomparison/api-diff.html)


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812737&view=ms.vss-test-web.test-result-details) on Agent 
Merge 8a1f1c9b34237188ab8c421deb2e3218f4de1880 into 165941ec143c8e71308a95e26e2de0fc0af4f77d

spouliot commented on runtime/runtime.m:
```diff
@@ -270,8 +270,10 @@
 	} else {
-		for (int i = 0; i < index + 1; i++)
+		for (int i = 0; i < index + 1; i++) {
+			xamarin_mono_object_release (&p);
```
I assume it's fine if the first iteration points to `NULL`, right ?

rolfbjarne replied:
Correct, a pointer to NULL will be ignored.

spouliot commented on runtime/runtime.m:
```diff
@@ -270,8 +270,10 @@
+		for (int i = 0; i < index + 1; i++) {
+			xamarin_mono_object_release (&p);
 			p = mono_signature_get_params (msig, &iter);
+		}
```
what will release the last `p` assigned in the loop ?

rolfbjarne replied:
It will be returned from this function, and the caller will have to release it.

rolfbjarne commented:
/azp run

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812973&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.16/4812973/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.16/4812973/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/ad17755850b00efdf39a71c89c89043f)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.16/4812973/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/f5753b2e10c03b4a6945076f13f7d3d4)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.16/4812973/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/23ce6adaf92ec76f95b063ed6bd55084)
* macCatiOS  [gist](https://gist.github.com/1002e6ab3774a675219618185561acf0)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.16/4812973/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/a77991f8a2fe1d673302b51c593946e1)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.16/4812973/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/b0ebf160c03906dc3c1aaadc7ddca989)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.16/4812973/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.16/4812973/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11730/4812973/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11730/4812973/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11730/4812973/apicomparison/api-diff.html)


# Test results

<details>
<summary>2 tests failed, 106 tests passed.</summary>

## Failed tests

 * monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Failed (Test run failed.
Tests run: 2444 Passed: 2353 Inconclusive: 7 Failed: 4 Ignored: 87)
 * monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Failed (Test run failed.
Tests run: 2441 Passed: 2353 Inconclusive: 7 Failed: 2 Ignored: 86)
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812973&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1100.BigSur
Merge 8a1f1c9b34237188ab8c421deb2e3218f4de1880 into 51c936457d6ce0f7f596179f0fbe6b294c56f6c7

rolfbjarne commented:
Test failures are unrelated

* monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Fixed in #11732, #11733 and #11734.
* monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Fixed in #11732 and #11733.
